doc/ref/chap-type-method.md

```diff
@@ -27,6 +27,25 @@ This chapter in the [Oils Reference](index.html) describes YSH types and methods
 
 ### replace()
 
+Replace substrings or eggex matches within a string.
+
+    var mystr = "code is 1234"
+    = mystr=>replace("code", "Code")  # => "Code is 1234"
+    = mystr=>replace(/ d+ /, "<redacted>")  # => "code is <redacted>"
+
+Replacement expressions can use captures from the given eggex.
+
+    var mystr = "year is 1988"
+    = mystr=>replace(/ <capture digit{4} as year : int> /, ^"$[year + 1]")
+    # => "year is 1989"
+
+To limit the number of replacements, pass in a named count argument. By default
+the count is `-1` which means "replace all".
+
+    var mystr = "bob has a friend named bob"
+    = mystr=>replace("bob", "Bob", count=1)   # => "Bob has a friend named bob"
+    = mystr=>replace("bob", "Bob", count=-1)  # => "Bob has a friend named Bob"
```

Comment:
Weirdly Python says that the count must be non-negative, and 0 means replace ALL

not replace zero!

I'm wondering if they have any reason for that, or it's just that people think replacing zero is useless.  But I think it could be consistent for some algorithms, not sure.

---

https://docs.python.org/3/library/re.html#re.sub

> The optional argument count is the maximum number of pattern occurrences to be replaced; count must be a non-negative integer